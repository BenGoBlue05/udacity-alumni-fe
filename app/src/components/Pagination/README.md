## Pagination Component
A reusable pagination component.

### Example

```js
<Pagination
   onChange={(newPage) => onChange(newPage)}
   defaultCurrent={1}
   pageSize={pageSize}
   current={currentPage}
   total={total}
 />
```
currentPage: PropTypes.number.isRequired,
total: PropTypes.number.isRequired,
onChange: PropTypes.func.isRequired,
### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **currentPage**    | Number   |             | Any Number value
| **total**    | Number   |             | Any Number value
| **onChange**    | Func   |             | Any function


### Other Information
Uses the rc-pagination component.  See here: https://github.com/react-component/pagination
