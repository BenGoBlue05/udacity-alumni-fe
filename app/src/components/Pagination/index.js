import React, { PropTypes } from 'react';
import Footer from 'grommet-udacity/components/Footer';
import RCPagination from 'rc-pagination';

const Pagination = ({
  onChange,
  pageSize,
  currentPage,
  total,
}) => (
  <Footer align="center" justify="center" pad="medium">
    <RCPagination
      style={{ color: 'white' }}
      onChange={(newPage) => onChange(newPage)}
      defaultCurrent={1}
      pageSize={pageSize}
      current={currentPage}
      total={total}
    />
  </Footer>
);

Pagination.propTypes = {
  onChange: PropTypes.func.isRequired,
  pageSize: PropTypes.number.isRequired,
  currentPage: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
};

Pagination.defaultProps = {
  pageSize: 10,
};

export default Pagination;
