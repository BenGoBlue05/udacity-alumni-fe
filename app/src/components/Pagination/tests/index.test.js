import Pagination from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<Pagination />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Pagination />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
