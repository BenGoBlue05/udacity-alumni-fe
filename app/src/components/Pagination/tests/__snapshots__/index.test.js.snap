exports[`<Pagination /> should render with default props 1`] = `
<Footer
  align="center"
  direction="row"
  justify="center"
  pad="medium"
  responsive={false}>
  <Pagination
    className=""
    defaultCurrent={1}
    defaultPageSize={10}
    locale={
      Object {
        "items_per_page": "条/页",
        "jump_to": "跳至",
        "next_5": "向后 5 页",
        "next_page": "下一页",
        "page": "页",
        "prev_5": "向前 5 页",
        "prev_page": "上一页"
      }
    }
    onChange={[Function onChange]}
    onShowSizeChange={[Function noop]}
    pageSize={10}
    prefixCls="rc-pagination"
    selectComponentClass={null}
    selectPrefixCls="rc-select"
    showQuickJumper={false}
    showSizeChanger={false}
    style={
      Object {
        "color": "white"
      }
    }
    total={0} />
</Footer>
`;
